compiler/luci/pass/src/FusePreActivationBatchNormPass.cpp

```diff
@@ -299,11 +299,81 @@ luci::CircleSub *insert_sub(luci::CircleNode *pred, luci::CircleConst *beta)
   return sub;
 }
 
+luci::CircleAdd *get_forward_add(luci::CircleNode *node)
+{
+  auto opcode = node->opcode();
+  if (opcode == luci::CircleOpcode::ADD)
+  {
+    auto add = loco::must_cast<luci::CircleAdd *>(node);
+    return add;
+  }
+  else if (opcode == luci::CircleOpcode::MAX_POOL_2D)
+  {
+    auto succ = loco::succs(node);
+    if (succ.size() != 1)
+      return nullptr;
```

Comment:
Just curiosity, is there `MAX_POOL_2D` that has two or more outputs?

Reply:
`MAX_POOL_2D` always has one output (i.e., produces one output tensor). The code is checking the number of successors that use the output.

Reply:
```
    auto succ = loco::succs(node);
    if (succ.size() != 1)
      return nullptr;
```
FYI, `succ` holds nodes that use `node` as input